HeroVault/HeroVault/Features/Home/HomeScreen/HomeScreenViewModel.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  HomeScreenViewModel.swift
+//  HeroVault
+//
+//  Created by Gia Catano on 2024/03/30.
+//
+
+import Foundation
+
+protocol ViewModelProtocol: AnyObject {
+    func reloadView()
+}
+
+class HomeScreenViewModel {
+    
+    // MARK: Variables
+    
+    var marvelData = [MarvelData]()
+    var filteredMarvelData = [MarvelData]()
+    var marvelDataType: EntityType
```

Comment:
These variables can probably be private right?

Reply:
yeah agreed
  private var marvelData = [MarvelData]()
    private var filteredMarvelData = [MarvelData]()
    private var marvelDataType: EntityType
    private var error: Error?
    private var isSearching: Bool
    private let homeScreenRepository: HomeScreenRepositoryType?
    private weak var delegate: ViewModelProtocol?
    var hideNoResultsText: Bool